File: leetcode/python/53.py
## Leetcode 53 - Maximum Subarray
## Nov. 27, 2023

from typing import List

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_arr, max_sum = nums[0], 0
        for i in nums:
            max_sum = max(0, max_sum) + i
            max_arr = max(max_arr, max_sum)
        return max_arr

File: leetcode/c/53.c
// Leetcode 53 - Maximum Subarray
// Nov. 27, 2023

int maxSubArray(int* nums, int numsSize) {
    int max_arr = nums[0], max_sum = 0;
    for (int i = 0; i < numsSize; i++) {
        max_sum = max_sum > 0 ? max_sum + nums[i] : nums[i];
        max_arr = max_sum > max_arr ? max_sum : max_arr;
    }
    return max_arr;
}